app/src/main/java/com/github/libretube/util/LinkHandler.kt

```diff
@@ -0,0 +1,54 @@
+package com.github.libretube.util
+
+import android.text.Editable
+import android.text.Spanned
+import android.text.TextPaint
+import android.text.style.ClickableSpan
+import android.view.View
+import com.github.libretube.util.HtmlParser.Companion.getValue
+import org.xml.sax.Attributes
+
+class LinkHandler(private val clickCallback: ((String) -> Unit)?) : HtmlParser.TagHandler {
+    private var linkTagStartIndex = -1
+    private var link: String? = null
+    override fun handleTag(
+        opening: Boolean,
+        tag: String?,
+        output: Editable?,
+        attributes: Attributes?
+    ): Boolean {
+        if (output != null) {
```

Comment:
These if conditions should be replaced with with some return statements instead.